.github/workflows/test-all-deps.yml

```diff
@@ -44,13 +44,6 @@ jobs:
       - run: pip install .
       - run: pip install matplotlib
       - run: pytest --ignore=test/install --cov=fairlearn --cov-report=xml test/
-      - name: Upload coverage to Codecov
+      - name: Upload coverage reports to Codecov with GitHub Action
         if: ${{ (inputs.codeCovPython == inputs.python) && (steps.get_date.outputs.AGE < 172800) }} # 172800 = 48 * 3600
-        uses: codecov/codecov-action@v3
-        with:
-          env_vars: OS,PYTHON
-          fail_ci_if_error: true
-          files: ./coverage.xml
-          flags: unittests
-          name: codecov-umbrella
-          verbose: true
```

Comment:
I'm surprised that none of these are needed / they don't set the correct behavior

Reply:
would you mind checking if removing them impacts the behaviour on the account since you know what the desired one is? or you are saying that they don't? I changed the GitHub action to a different one so it might mean there is some behaviour included that wasn't there before?

Reply:
Basically, these settings need to be either on the codecov's config (codecov.yml) file or in the CI, and right now these settings here are not present in the config file, therefore this is changing behavior.

Reply:
ok, will do! the only one that can't be used at the moment is `path_to_write_report` which is deprecated. Is that one also strictly necessary? found only here https://github.com/fairlearn/fairlearn/pull/1399/files#diff-453070e1c523e30f2e01a1e4715f6ab1e5a4bc94cd0bd73234c8a2952bb33f46L51

Reply:
no, remove if deprecated.

Reply:
removed in https://github.com/fairlearn/fairlearn/pull/1399/commits/de2049457e29e4004ebc1a530544320230144d80, we just need a confirmation now that CI passes as before

Reply:
CI is green 🥳